stellargraph/core/graph.py

```diff
@@ -147,6 +147,34 @@ def __init__(
             dtype,
         )
 
+    # customise how a missing attribute is handled to give better error messages for the NetworkX
+    # -> no NetworkX transition.
+    def __getattr__(self, item):
+        import networkx
+
+        try:
+            # get the native python wording of the error message
+            super().__getattribute__(item)
+        except AttributeError as e:
+            if hasattr(networkx.MultiDiGraph, item):
+                # a networkx class has this as an attribute, so let's assume that it's old code
+                # from before the conversion and replace (the `from None`) the default exception
+                # with one with a more specific message that guides the user to the fix
+                type_name = type(self).__name__
+                raise AttributeError(
+                    f"{e.args[0]}. The '{type_name}' type no longer inherits from NetworkX types: use a new StellarGraph method, or, if that is not possible, the `.to_networkx()` conversion function."
+                ) from None
+
+            # doesn't look like a NetworkX method so use the default error
+            raise
+        else:
+            # __getattr__ is only called for attribute accesses that don't exist,
+            # so the `__getattribute__` call should also fail, but, just in case it doesn't,
+            # let's not return None implicitly
+            raise AssertionError(
+                f"attribute '{item}' that was missing has magically appeared again"
```

Comment:
I'm thinking this should only happen when someone directly calls `graph.__getattr__('missing_field')` so the error message could talk about that instead if you want

Reply:
I'm not sure I understand? Do you mean that this code path can only happen for a direct call like that? Or do you think that we should detect whether it's a direct call and only do this then? Or something else?

If the first one, that's not true, e.g. the test shows that `some_stellargraph.successors` takes this code path, and so would `getattr(some_stellargraph, "successors")`?

Reply:
Oops sorry I mean to say `graph.__getattr__('existing_field')`, and was referring to the code path that results in this message about "attribute that was missing has magically appeared again". 
```
>>> class A:
...   def __init__(self, x):
...     self.x = x
...   def __getattr__(self, item):
...     try:
...       super().__getattribute__(item)
...     except AttributeError as e:
...       print(e)
...       raise
...     else:
...       print("hmm")
...       raise
...
>>> a = A(1)
>>> a.x
1
>>> a.y
'A' object has no attribute 'y'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 6, in __getattr__
AttributeError: 'A' object has no attribute 'y'
>>> a.__getattr__('x')
hmm
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 12, in __getattr__
RuntimeError: No active exception to reraise
>>>
```

Reply:
Oh, I see. Yes, it seems this case isn't actually impossible...